src/main/java/javax/jmdns/impl/DNSQuestion.java

```diff
@@ -24,6 +24,28 @@
 public class DNSQuestion extends DNSEntry {
     private static final Logger logger = LoggerFactory.getLogger(DNSQuestion.class);
 
+    /**
+     * Checks if the local host name matches the name of this instance, ignoring case.
+     * If there is a match, adds the DNS records for the local host to the provided set of answers.
+     *
+     * @param jmDNSImpl The {@link JmDNSImpl} instance representing the local DNS service.
+     * @param answers   The set of {@link DNSRecord} to which the local host's DNS records should be added if a match is found.
+     * @return {@code true} if the local host name matches this instance's name (case-insensitive); {@code false} otherwise.
+     */
+    boolean isHostMatch(JmDNSImpl jmDNSImpl, Set<DNSRecord> answers) {
+        HostInfo hostInfo = jmDNSImpl.getLocalHost();
+        String localHostName = hostInfo.getName();
+
+        // Use equalsIgnoreCase directly without lowercasing.
+        if (!localHostName.equalsIgnoreCase(this.getName())) {
+            return false;
+        }
+
+        // If the names match, add all the answers and return true.
+        answers.addAll(hostInfo.answers(this.getRecordClass(), this.isUnique(), DNSConstants.DNS_TTL));
```

Comment:
This is imho an unexpected sideeffect of the method. I expect an `isXYZ`method to do the check and return `true` or `false`, but be without any additional effects.

Reply:
Sure thing let me try and improve this i was trying to DRY it up with changing as little code as possible.  But let me see if it can be improved.

Reply:
You are right i like the code but renamed `isHostCheck` to `processHostMatch` to indicate its more than a boolean that is processing something,